queue/queue.go

```diff
@@ -2,23 +2,26 @@ package queue
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strings"
 
 	"github.com/nats-io/nats.go"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
+	"golang.org/x/exp/slices"
 
 	"github.com/connylabs/ingest"
 )
 
 type queue struct {
 	js                          nats.JetStreamContext
 	conn                        *nats.Conn
-	reg                         prometheus.Registerer
 	queueOperationsTotalCounter *prometheus.CounterVec
 }
 
 // New is able to connect to the queue
-func New(url string, reg prometheus.Registerer) (ingest.Queue, error) {
+func New(url string, stream, subject string, reg prometheus.Registerer) (ingest.Queue, error) {
```

Comment:
I think maybe we should change the signature so that the caller doesn't expect the queue constructor to do some magic on the stream. Instead the caller should pass exactly the slice that it expects. WDYT?

Reply:
done